packages/preferences-api/src/common/preference-service.ts

```diff
@@ -65,7 +66,7 @@ export class PreferenceService implements Disposable {
     get<T>(preferenceName: string, defaultValue: T): T;
     get<T>(preferenceName: string, defaultValue?: T): T | undefined {
         const value = this.prefCache[preferenceName];
-        return value !== null && value !== undefined ? value : defaultValue;
+        return value !== null && value !== undefined ? clone(value) : clone(defaultValue);
```

Comment:
looking at https://github.com/theia-ide/theia/blob/master/packages/preferences-api/src/browser/preference-proxy.ts#L65 again, not all values returned by a preference proxy come from here, we have to clone there as well